include/openPMD/auxiliary/StringManip.hpp

```diff
@@ -126,8 +126,9 @@ join(std::vector< std::string > const& vs, std::string const& delimiter)
         default:
             std::ostringstream ss;
             std::copy(vs.begin(),
-                      vs.end(),
+                      vs.end() - 1,
```

Comment:
o.0

this changes looks a bit unsave, e.g. for zero-lengths :)

Reply:
The empty and size 1 array cases are handled above ;)

Reply:
yep, saw this too late :)